# Profiling avy_obs_pkl

In [1]:
import pandas as pd
import pandas_profiling

avy_obs = pd.read_pickle('pkl/avy_obs.pkl')
avy_obs

avpathid  avobsdate avobstime avobserverid avtype  avtriggertyp  \
avrid                                                                    
1        SHED10 2012-03-26  12:45:00     stein_01     WS           5.0   
2      PATH1163 2012-03-16  08:30:00     stein_01      U           5.0   
3         CURLY 2012-03-16  08:34:00     stein_01      U           5.0   
4        SHED10 2012-03-10  14:50:00     stein_01     WL           5.0   
5      INFINITY 2012-03-10  14:54:00     stein_01      L           5.0   
...         ...        ...       ...          ...    ...           ...   
504     SHED07W 2022-03-01  05:04:00     clark_01     SS           5.0   
505     AP1163H 2022-03-23  13:30:00     clark_01     WL           5.0   
506      SHED10 2022-03-23  13:30:00     clark_01     WL           5.0   
507     MCAMSER 2022-03-23  18:00:00     clark_01     HS           5.0   
508    PATH1163 2022-03-27  14:35:00     clark_01     WL           5.0   

      avtrigger avrelativesize avdestructivesize avinterface  ... deb_hwy  \
avrid                                                         ...           
1            NL             R1              D1.5           O  ...   False   
2            NU              U                D2           U  ...   False   
3            NU              U                D1           U  ...   False   
4           NaN             R2                D1           I  ...   False   
5             N             R2                D1           I  ...   False   
...         ...            ...               ...         ...  ...     ...   
504          NC             R2                D2           I  ...   False   
505          NU             R1              D1.5           O  ...   False   
506           N             R2                D2           O  ...   False   
507          NC             R3                D3           O  ...   False   
508           N             R1                D2           O  ...   False   

      deb_hwy_w  deb_hwy_d                                          avcomment  \
avrid                                                                           
1           0.0        0.0  Wet slab released on SE aspect @ 5400 feet (16...   
2           0.0        0.0  Wet debris ran 2/3 path in main path to 4700 f...   
3           0.0        0.0  Terminus at canyon floor elevation.\r\n\r\nDeb...   
4           0.0        0.0                                               D1.5   
5           0.0        0.0                                    D1.5- 1/2 Path.   
...         ...        ...                                                ...   
504         0.0        0.0  Slab not observed until March 17th during a sk...   
505         0.0        0.0                                                NaN   
506         0.0        0.0                          Obs on Mar. 24th from Hwy   
507         0.0        0.0  Obs on Mar. 24th from Hwy. Could not see runou...   
508         0.0        0.0                        Obs. on Mar. 28th from Hwy.   

      edcr_user           edcr_time       edcr_ip  eded_user  \
avrid                                                          
1            ed 2016-03-17 11:50:56  10.173.132.2        NaN   
2            ed 2016-03-17 11:50:56  10.173.132.2        NaN   
3            ed 2016-03-17 11:50:56  10.173.132.2        NaN   
4            ed 2016-03-17 11:50:56  10.173.132.2        NaN   
5            ed 2016-03-17 11:50:56  10.173.132.2        NaN   
...         ...                 ...           ...        ...   
504        Adam 2022-03-17 21:17:04    10.30.72.4        NaN   
505        Adam 2022-03-23 21:32:42    10.30.72.4        NaN   
506        Adam 2022-03-28 16:30:35    10.30.72.4        NaN   
507        Adam 2022-03-28 16:34:43    10.30.72.4        NaN   
508        Adam 2022-03-28 16:37:03    10.30.72.4        NaN   

                 eded_time  eded_ip  
avrid                                
1      0000-00-00 00:00:00      NaN  
2      0000-00-00 00:00:00      NaN  
3  

In [2]:
profile = avy_obs.profile_report(title = 'Avalanche Observations Profiling')
profile.to_file(output_file='data-profiles/avy_obs-profile.html')

Summarize dataset:   0%|          | 0/53 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

# Profile Findings

#### The following 18 columns have missing values: 
    avtype has 6 (1.2%) missing values	
    avtriggertyp has 36 (7.3%) missing values	
    avtrigger has 45 (9.2%) missing values	
    avrelativesize has 17 (3.5%) missing values	
    avdestructivesize has 50 (10.2%) missing values	
    avinterface has 60 (12.2%) missing values	
    sz_vert has 21 (4.3%) missing values	
    sz_horiz has 38 (7.7%) missing values	
    sz_aspect has 386 (78.6%) missing values	
    term_longpath has 30 (6.1%) missing values	
    term_shortPath has 165 (33.6%) missing values	
    term_detail has 184 (37.5%) missing values	
    term_moist has 113 (23.0%) missing values	
    avcomment has 129 (26.3%) missing values	
    edcr_user has 127 (25.9%) missing values	
    edcr_ip has 98 (20.0%) missing values	
    eded_user has 294 (59.9%) missing values
    eded_ip has 293 (59.7%) missing values